unstructured_ingest/v2/processes/connectors/outlook.py

```diff
@@ -0,0 +1,218 @@
+import hashlib
+import time
+from dataclasses import dataclass, field
+from datetime import timezone
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, Coroutine, Generator
+
+from pydantic import Field, Secret
+
+from unstructured_ingest.error import SourceConnectionError
+from unstructured_ingest.logger import logger
+from unstructured_ingest.utils.dep_check import requires_dependencies
+from unstructured_ingest.v2.interfaces import (
+    AccessConfig,
+    ConnectionConfig,
+    Downloader,
+    DownloaderConfig,
+    FileData,
+    Indexer,
+    IndexerConfig,
+    download_responses,
+)
+from unstructured_ingest.v2.interfaces.file_data import FileDataSourceMetadata, SourceIdentifiers
+from unstructured_ingest.v2.processes.connector_registry import SourceRegistryEntry
+
+MAX_EMAILS_PER_FOLDER = 1_000_000  # Maximum number of emails per folder
+
+if TYPE_CHECKING:
+    from office365.graph_client import GraphClient
+    from office365.outlook.mail.folders.folder import MailFolder
+    from office365.outlook.mail.messages.message import Message
+
+
+CONNECTOR_TYPE = "outlook"
+
+
+class OutlookAccessConfig(AccessConfig):
+    client_credential: str = Field(description="Microsoft App client secret", alias="client_cred")
+
+
+class OutlookConnectionConfig(ConnectionConfig):
+    access_config: Secret[OutlookAccessConfig]
+    client_id: str = Field(description="Microsoft App client ID")
+    tenant: str = Field(
+        default="common", description="ID or domain name associated with your Azure AD instance"
+    )
+    authority_url: str = Field(
+        default="https://login.microsoftonline.com",
+        description="Authentication token provider for Microsoft apps",
+    )
+
+    @requires_dependencies(["msal"])
```

Comment:
Is there an extra associated with this dependency that can be added here?

Reply:
No there isn't. `pip install msal` is sufficient.

However based on your question I've looked into the `requires_dependencies` and it seems that I it works in slightly different way then I though. In the other usage of `@requires_dependecies` in this file, which I've taken directly from V1, we have this snippet
```py
    @requires_dependencies(["office365"], extras="graph")
    @SourceConnectionError.wrap
    def get_client(self) -> "GraphClient":
        from office365.graph_client import GraphClient
```
if office365 is missing from dependencies it will suggest
```
ImportError: Following dependencies are missing: office365. Please install them using `pip install "unstructured-ingest[graph]"`.
Error: Following dependencies are missing: office365. Please install them using `pip install "unstructured-ingest[graph]"`.
```
which doesn't make sense. Even if interpreted as I initially did, I expected it to suggest installing `office365[graph]` that also wouldn't be accurate because we need to install `Office365-REST-Python-Client` actually.

Looks like the `requires_dependencies` decorator in its current form doesn't fit this case at all.